fish-rust/src/common.rs

```diff
@@ -344,15 +371,14 @@ fn escape_string_url(input: &wstr) -> WString {
 }
 
 /// Escape a string in a fashion suitable for using as a fish var name. Store the result in out_str.
-#[widestrs]
 fn escape_string_var(input: &wstr) -> WString {
     let mut prev_was_hex_encoded = false;
     let narrow = wcs2string(input);
     let mut out = WString::new();
-    for byte in narrow.into_iter() {
+    for byte in narrow {
         if (byte & 0x80) == 0 {
             let c = char::from_u32(u32::from(byte)).unwrap();
-            if c.is_alphanumeric() && (!prev_was_hex_encoded || c.to_digit(16).is_none()) {
+            if c.is_alphanumeric() {
```

Comment:
I mentioned this on gitter, the unescaping in Rust was bugged, example:

```
string escape --style=var 'a b#c"\'d'
  The CHECK on line 227 wants:
    a_20_b_23_c_22_27_d

  which failed to match line stdout:84:
    a_20_62_23_63_22_27_64
```

From comparing this to https://github.com/fish-shell/fish-shell/blob/a75de42f4b00d711f59e383a8a9f07944530ab4e/src/common.cpp#L744 I have the following takeaways:
1. I have no clue why we are checking if it's hexadecimal
2. I have no clue why the C++-version _did_ print out the chars without escaping them to their char codes, in the example above we must have that `swprintf ` is not following the format specifier? Since it will not go into the first case after handling the space

Reply:
Okay my confusion has been cleared up: C++ only transforms uppercase letters (A..=F) to hex, while lowercase are not, will correct the "fix" which causes behaviour issues and add tests

Reply:
Moved on to #9869